File: tests/algorithms/x/Java/data_structures/heap/max_heap.java
public class Main {
    static int[] heap = new int[0];
    static int size = 0;

    static void swap_up(int i) {
        int temp = heap[i];
        int idx = i;
        while (idx / 2 > 0) {
            if (heap[idx] > heap[idx / 2]) {
heap[idx] = heap[idx / 2];
heap[idx / 2] = temp;
            }
            idx = idx / 2;
        }
    }

    static void insert(int value) {
        heap = ((int[])(java.util.stream.IntStream.concat(java.util.Arrays.stream(heap), java.util.stream.IntStream.of(value)).toArray()));
        size = size + 1;
        swap_up(size);
    }

    static void swap_down(int i) {
        int idx_1 = i;
        while (size >= 2 * idx_1) {
            int bigger_child = 2 * idx_1 + 1 > size ? 2 * idx_1 : heap[2 * idx_1] > heap[2 * idx_1 + 1] ? 2 * idx_1 : 2 * idx_1 + 1;
            int temp_1 = heap[idx_1];
            if (heap[idx_1] < heap[bigger_child]) {
heap[idx_1] = heap[bigger_child];
heap[bigger_child] = temp_1;
            }
            idx_1 = bigger_child;
        }
    }

    static void shrink() {
        int[] new_heap = ((int[])(new int[]{}));
        int i = 0;
        while (i <= size) {
            new_heap = ((int[])(java.util.stream.IntStream.concat(java.util.Arrays.stream(new_heap), java.util.stream.IntStream.of(heap[i])).toArray()));
            i = i + 1;
        }
        heap = ((int[])(new_heap));
    }

    static int pop() {
        int max_value = heap[1];
heap[1] = heap[size];
        size = size - 1;
        shrink();
        swap_down(1);
        return max_value;
    }

    static int[] get_list() {
        int[] out = ((int[])(new int[]{}));
        int i_1 = 1;
        while (i_1 <= size) {
            out = ((int[])(java.util.stream.IntStream.concat(java.util.Arrays.stream(out), java.util.stream.IntStream.of(heap[i_1])).toArray()));
            i_1 = i_1 + 1;
        }
        return out;
    }

    static int len() {
        return size;
    }
    public static void main(String[] args) {
        heap = ((int[])(new int[]{0}));
        size = 0;
        insert(6);
        insert(10);
        insert(15);
        insert(12);
        System.out.println(pop());
        System.out.println(pop());
        System.out.println(get_list());
        System.out.println(len());
    }
}

File: tests/algorithms/x/PHP/data_structures/heap/max_heap.php
<?php
ini_set('memory_limit', '-1');
function _append($arr, $x) {
    $arr[] = $x;
    return $arr;
}
function _intdiv($a, $b) {
    if (function_exists('bcdiv')) {
        $sa = is_int($a) ? strval($a) : (is_string($a) ? $a : sprintf('%.0f', $a));
        $sb = is_int($b) ? strval($b) : (is_string($b) ? $b : sprintf('%.0f', $b));
        return intval(bcdiv($sa, $sb, 0));
    }
    return intdiv($a, $b);
}
$heap = [0];
$size = 0;
function swap_up($i) {
  global $heap, $size;
  $temp = $heap[$i];
  $idx = $i;
  while (_intdiv($idx, 2) > 0) {
  if ($heap[$idx] > $heap[_intdiv($idx, 2)]) {
  $heap[$idx] = $heap[_intdiv($idx, 2)];
  $heap[_intdiv($idx, 2)] = $temp;
}
  $idx = _intdiv($idx, 2);
};
}
function insert($value) {
  global $heap, $size;
  $heap = _append($heap, $value);
  $size = $size + 1;
  swap_up($size);
}
function swap_down($i) {
  global $heap, $size;
  $idx = $i;
  while ($size >= 2 * $idx) {
  $bigger_child = (2 * $idx + 1 > $size ? 2 * $idx : ($heap[2 * $idx] > $heap[2 * $idx + 1] ? 2 * $idx : 2 * $idx + 1));
  $temp = $heap[$idx];
  if ($heap[$idx] < $heap[$bigger_child]) {
  $heap[$idx] = $heap[$bigger_child];
  $heap[$bigger_child] = $temp;
}
  $idx = $bigger_child;
};
}
function shrink() {
  global $heap, $size;
  $new_heap = [];
  $i = 0;
  while ($i <= $size) {
  $new_heap = _append($new_heap, $heap[$i]);
  $i = $i + 1;
};
  $heap = $new_heap;
}
function pop() {
  global $heap, $size;
  $max_value = $heap[1];
  $heap[1] = $heap[$size];
  $size = $size - 1;
  shrink();
  swap_down(1);
  return $max_value;
}
function get_list() {
  global $heap, $size;
  $out = [];
  $i = 1;
  while ($i <= $size) {
  $out = _append($out, $heap[$i]);
  $i = $i + 1;
};
  return $out;
}
function len() {
  global $heap, $size;
  return $size;
}
insert(6);
insert(10);
insert(15);
insert(12);
echo rtrim(json_encode(pop(), 1344)), PHP_EOL;
echo rtrim(json_encode(pop(), 1344)), PHP_EOL;
echo rtrim(str_replace('false', 'False', str_replace('true', 'True', str_replace('"', '\'', str_replace(':', ': ', str_replace(',', ', ', json_encode(get_list(), 1344))))))), PHP_EOL;
echo rtrim(json_encode(len(), 1344)), PHP_EOL;
